transactions/accounts/add_key.cdc

```diff
@@ -4,9 +4,20 @@ transaction(key: String, signatureAlgorithm: UInt8, hashAlgorithm: UInt8, weight
 
 	prepare(signer: auth(AddKey) &Account) {
 		pre {
-			signatureAlgorithm >= 1 && signatureAlgorithm <= 3: "Must provide a signature algorithm raw value that is 1, 2, or 3"
-			hashAlgorithm >= 1 && hashAlgorithm <= 6: "Must provide a hash algorithm raw value that is between 1 and 6"
-			weight <= 1000.0: "The key weight must be between 0 and 1000"
+			signatureAlgorithm >= 1 && signatureAlgorithm <= 3:
+                "Cannot add Key: Must provide a signature algorithm raw value that corresponds to "
+                .concat("one of the available signature algorithms for Flow keys.")
+                .concat("You provided ").concat(signatureAlgorithm.toString())
+                .concat(" but the options are either 1 (ECDSA_P256), 2 (ECDSA_secp256k1), or 3 (BLS_BLS12_381).")
+			hashAlgorithm >= 1 && hashAlgorithm <= 6:
+                "Cannot add Key: Must provide a hash algorithm raw value that corresponds to "
+                .concat("one of of the available hash algorithms for Flow keys.")
+                .concat("You provided ").concat(hashAlgorithm.toString())
+                .concat(" but the options are 1 (SHA2_256), 2 (SHA2_384), 3 (SHA3_256), ")
+                .concat("4 (SHA3_384), 5 (KMAC128_BLS_BLS12_381), or 6 (KECCAK_256).")
+			weight <= 1000.0:
+                "Cannot add Key: The key weight must be between 0 and 1000."
+                .concat(" You provided ").concat(weight.toString()).concat(" which is invalid.")
```

Comment:
These updated error messages are really helpful, but I'm starting to wonder if there's a way we can standardize some of them as they're getting pretty long. Maybe something like `FlowCoreErrors` and in this instance `invalidSigAlgo(_ provided: UInt8): String` and `invalidHashAlgo(_ provided: UInt8): String`. Modifiers would be nice, but that's a whole new language feature.

Reply:
It is a little awkward right now, but once we have better string formatting in Cadence, then I think it won't be as bad. I think having a separate contract for it is a little unnecessary.